components/validator/challenger.go

```diff
@@ -365,87 +397,128 @@ func (c *Challenger) handleOutput(ctx context.Context, outputIndex *big.Int) {
-
-			// check the status of challenge
-			status, err := c.GetChallengeStatus(outputIndex)
+			isOutputFinalized, err := c.IsOutputFinalized(c.ctx, outputIndex)
 			if err != nil {
-				c.log.Error("unable to get challenge status", "err", err, "outputIndex", outputIndex)
+				c.log.Error("unable to get if output is finalized when handling challenge", "err", err, "outputIndex", outputIndex)
 				continue
 			}
 
-			// if the challenge is inactivated, terminate handling
-			if isInactivated(status) {
-				c.log.Error("challenge is not in progress", "challengeStatus", status)
-				return
-			}
-
 			// if asserter
-			if isAsserter && c.cfg.OutputSubmitterEnabled {
-				if status == chal.StatusAsserterTurn {
-					tx, err := c.Bisect(ctx, outputIndex)
+			if isAsserter {
+				// if output is already finalized, terminate handling
+				if isOutputFinalized {
+					c.log.Info("asserter: output is already finalized when handling challenge", "outputIndex", outputIndex, "challenger", challenger)
+					return
+				}
+
+				// if output is already deleted, asserter has no incentives to handle challenge any further
+				if isOutputDeleted {
+					c.log.Info("asserter: do nothing because output is already deleted", "outputIndex", outputIndex, "challenger", challenger)
+					return
+				}
+
+				switch status {
+				case chal.StatusAsserterTurn:
+					tx, err := c.Bisect(c.ctx, outputIndex, challenger)
+					if err != nil {
+						c.log.Error("asserter: failed to create bisect tx", "err", err, "outputIndex", outputIndex, "challenger", challenger)
+						continue
+					}
+					if err := c.submitChallengeTx(tx); err != nil {
+						c.log.Error("asserter: failed to submit bisect tx", "err", err, "outputIndex", outputIndex, "challenger", challenger)
+						continue
+					}
+				case chal.StatusChallengerTimeout:
+					// call challenger timeout to increase bond from pending bond
+					tx, err := c.ChallengerTimeout(c.ctx, outputIndex, challenger)
 					if err != nil {
-						c.log.Error("asserter: failed to create bisect tx", "err", err, "outputIndex", outputIndex)
+						c.log.Error("asserter: failed to create challenger timeout tx", "err", err, "outputIndex", outputIndex, "challenger", challenger)
 						continue
 					}
-					if err := c.submitChallengeTx(ctx, tx); err != nil {
-						c.log.Error("asserter: failed to submit bisect tx", "err", err, "outputIndex", outputIndex)
+					if err := c.submitChallengeTx(tx); err != nil {
+						c.log.Error("asserter: failed to submit challenger timeout tx", "err", err, "outputIndex", outputIndex, "challenger", challenger)
 						continue
 					}
 				}
 			}
 
 			// if challenger
 			if isChallenger && c.cfg.ChallengerEnabled {
+				// if output is already finalized, terminate handling
+				if isOutputFinalized {
+					// if output has been deleted when finalized, cancel challenge to refund pending bond
+					if status != chal.StatusChallengerTimeout && isOutputDeleted {
```

Comment:
why not explicitly modify the conditional?
```suggestion
					if status == chal.StatusAsserterTimeout && isOutputDeleted {
```

Reply:
This is because the `_cancelIfOutputDeleted` function in the Colosseum contract checks the same condition as above.